aptos-move/block-executor/src/executor.rs

```diff
@@ -359,14 +359,16 @@ where
     pub fn execute_transactions_parallel(
         &self,
         executor_initial_arguments: E::Argument,
-        signature_verified_block: Vec<T>,
+        signature_verified_block: &Vec<T>,
     ) -> Result<
         (
             Vec<E::Output>,
             OutputDeltaResolver<<T as Transaction>::Key, <T as Transaction>::Value>,
         ),
         E::Error,
     > {
+        assert!(self.concurrency_level > 1, "Must use sequential execution");
```

Comment:
Was wondering if those assertion would fail in production as they are a good source for the network availability attack. Would it be better to return an error when such condition is violated?

Reply:
Currently, must be true due to 
https://github.com/aptos-labs/aptos-core/blob/e197a64f990b349b888eec4624c1adf945d0ef67/aptos-move/aptos-vm/src/block_executor/mod.rs#L150, and second follow-up PR moves this dispatching inside block_executor crate: https://github.com/aptos-labs/aptos-core/blob/b423013ebba88a6362add5aa0a6077fe01edcb41/aptos-move/block-executor/src/executor.rs#L377, and execute_parallel becomes pub(crate), so we could also just delete the assert at that point. In fact, I like that idea to delete just this assert then.

If you really think we should return an error here, let me know what kind of error and what should happen in that case, we skip the whole block? But it will complicate some places and we should worry about determinism and probably not worth for some invariant that trivially should never happen.